EIPS/eip-7378.md

```diff
@@ -0,0 +1,81 @@
+---
+eip: 7378
+title: Add time-weighted averaging to the base fee
+description: Using geometric weights to average past block sizes into consideration
+author: Guy Goren (@guy-goren) <guy.nahsholim@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/add-time-weighted-averaging-to-the-base-fee-mechanism/15142
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-07-22
+---
+
+## Abstract
+
+This EIP proposes a new formula for the base fee, derived from [EIP-1559](./eip-1559.md). The existing base fee formula,
+
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s[i]-s^* }{s^* }\right)$$
+
+only considers the last block size $s[i]$. This mechanism incentivizes proposers to collude with users to manipulate the base fee.
+
+We propose that previous block sizes be considered by replacing the previous block size with an exponential moving average. In particular, we suggest the following formula:
+
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s_{\textit{avg}}[i]-s^* }{s^* }\right)$$
+
+where $s_{\textit{avg}}[i]$ is defined by:
+
+$$s_{\textit{avg}}[i] \triangleq \alpha\sum_{k=1}^{\infty} (1-\alpha)^k\cdot s[i-k+1]$$ 
+
+and $\alpha\in(0,1)$ is a smoothing factor.
+
+## Motivation
+
+Mainly, to reduce bribe motivation (in particular in case the demand for blockspace is high -- see Incentive Considerations section). Additionaly, to reduce oscillations and have a more stable fee setting mechanism.
+
+Proposers use a mechanism described in EIP-1559 to determine which messages to include in a block. This mechanism includes a "base fee": a portion of the transaction fee that is burned. The base fee varies according to the fill rate of blocks. A target block size is defined. If a block exceeds the target size, the base fee increases, and if it is smaller, the base fee lowers.
```

Comment:
can this be shortned assuming the audience is a bit aware of 1559

Reply:
It could certainly be omitted assuming the reader is familiar with the base fee mechanism. However, I believe that dedicating a few lines (approximately three) is reasonable to accommodate readers who may be less familiar with the topic.